### PR TITLE
Fix unicode issue with SORTKEY on python 2

### PR DESCRIPTION
**Pull request by bouk**
This is to make sure unicode is supported in python2

@graingert

### CODE REVIEW

File: tests/test_ddl_compiler.py [hunk 1]
```diff
@@ -131,6 +132,25 @@ def test_create_table_with_sortkey(self, compiler):
         )
         assert expected == actual, self._compare_strings(expected, actual)
 
+    @pytest.mark.skipif(sys.version_info[0] != 2, reason="only applicable to python 2")
+    def test_create_table_with_unicode_sortkey(self, compiler):
+        table = Table('t1',
+                      MetaData(),
+                      Column('id', Integer, primary_key=True),
+                      Column('name', String),
+                      redshift_sortkey=unicode("id"))
```
Comment (graingert) [inline tests/test_ddl_compiler.py]: You can use u'id' and this will work on both

File: tests/test_ddl_compiler.py [hunk 2]
```diff
@@ -131,6 +132,25 @@ def test_create_table_with_sortkey(self, compiler):
         )
         assert expected == actual, self._compare_strings(expected, actual)
 
+    @pytest.mark.skipif(sys.version_info[0] != 2, reason="only applicable to python 2")
```
Comment (jklukas) [inline tests/test_ddl_compiler.py]: Line too long here.

File: tests/test_ddl_compiler.py (hunk 1 above)
Comment (bouk) [inline tests/test_ddl_compiler.py]: fixed

Comment (jklukas) [inline tests/test_ddl_compiler.py]: The `lint` test runs on python 3 and throws an error for this `unicode`.

**Comment by jklukas** [comment]
+1

File: redshift_sqlalchemy/dialect.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
 from sqlalchemy.ext.compiler import compiles
 from sqlalchemy.sql.expression import Executable, ClauseElement
 from sqlalchemy.types import VARCHAR, NullType
+from sqlalchemy.util.compat import string_types
```
Comment (bouk) [inline redshift_sqlalchemy/dialect.py]: hmm looking at it it's definitely not public. What would be the best way to go about this? If I were to copy over the code, the linter would complain about unicode or basestring

Comment (graingert) [inline redshift_sqlalchemy/dialect.py]: What about a compat.py with a # noqa line?

**Comment by bouk** [comment]
@graingert how's this look?

File: redshift_sqlalchemy/compat.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+import sys
+
+
+py3k = sys.version_info >= (3, 0)
+
+if py3k:
+    string_types = str,
+else:
+    string_types = basestring,
```
Comment (graingert) [inline redshift_sqlalchemy/compat.py]: I prefer parenthesis with my tuples

Comment (graingert) [inline redshift_sqlalchemy/compat.py]: can you move the noqa to this line?

Comment (bouk) [inline redshift_sqlalchemy/compat.py]: Me too, I just copied the sa style. fixed

**Comment by graingert** [comment]
Looks good to me, have a look at the fixes then update the issue name, changelog, squash etc

**Comment by bouk** [comment]
done

**Pull request merged**